experimental/auth-logic/top_level.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"github.com/project-oak/transparent-release/experimental/auth-logic/wrappers"
+	"strings"
+)
+
+func VerifyRelease(appName string, endorsementFilePath string,
+	provenanceFilePath string) (string, error) {
+
+	endorsementAppName, err := wrappers.GetAppNameFromEndorsement(endorsementFilePath)
+	if err != nil {
+		fmt.Errorf("couldn't get name from endorsement file: %s, error: %v",
+			endorsementFilePath, err)
+	}
+	endorsementStatement, err := wrappers.EmitStatementAs(
+		wrappers.Principal{
+			Contents: fmt.Sprintf("%s::EndorsementFile", endorsementAppName),
```

Comment:
For predicate arguments and principal names, quotes mean that it is a constant rather than a variable. Right this was a mistake because it's supposed to be a constant. 

I think "type safe" is not really the right way to describe this -- we wouldn't solve this problem with a type system because it can be handled at the syntax level before the AST is even generated. I see what you mean though, maybe this syntax is not intuitive; I just replicated Souffle's syntax to get a prototype running quickly. I think Gogul wants to iterate on the language design anyway.